server.py

```diff
@@ -134,44 +134,46 @@ def validate():
             if schema is None:
                 return client_error("Unsupported schema version '%s'" % version)
 
-            bound_logger = logger.bind(
-                survey_id=json_data.get("survey_id"), tx_id=json_data.get("tx_id")
-            )
+            metadata = json_data.get("metadata")
+            bound_logger = logger.bind(survey_id=json_data.get("survey_id"),
+                                       tx_id=json_data.get("tx_id"),
+                                       user_id=metadata.get("user_id"),
+                                       ru_ref=metadata.get("ru_ref"))
 
             bound_logger.debug("Validating json against schema")
             schema(json_data)
 
             survey_id = json_data.get("survey_id")
             if survey_id not in KNOWN_SURVEYS.get(version, {}):
                 bound_logger.debug("Survey id is not known", survey_id=survey_id)
-                return client_error("Unsupported survey '%s'" % survey_id)
+                return client_error(f"Unsupported survey '{survey_id}'")
 
             instrument_id = json_data.get("collection", {}).get("instrument_id")
             if instrument_id not in KNOWN_SURVEYS.get(version, {}).get(survey_id, []):
                 bound_logger.debug("Instrument ID is not known", survey_id=survey_id)
-                return client_error("Unsupported instrument '%s'" % instrument_id)
+                return client_error(f"Unsupported instrument '{instrument_id}'")
 
-            metadata = json_data.get("metadata")
-            bound_logger.debug(
-                "Success",
-                user_id=metadata.get("user_id"),
-                ru_ref=metadata.get("ru_ref"),
-            )
+            if json_data.get('data'):
+                if json_data['version'] == '0.0.1':
+                    for value in json_data['data'].values():
+                        if r'\u0000' in str(value):
+                            bound_logger.debug("Null character found in submission")
```

Comment:
I think this should be an error or at least warn log message

Reply:
Please could this be WARN? It should only be ERROR if we're expecting ops to investigate it

Reply:
I'll put it as an error as the submission will need to be investigated if it fails validation